src/test/java/com/puppycrawl/tools/checkstyle/PropertyCacheFileTest.java

```diff
@@ -284,4 +292,200 @@ public void testChangeInConfig() throws Exception {
+        // throw CheckstyleException with the specific content.
+        try (MockedStatic<CommonUtil> commonUtil = mockStatic(CommonUtil.class)) {
+            final CheckstyleException mockException = new CheckstyleException(
+                    "Cannot get URL for cache file " + cacheFile.getAbsolutePath());
+            commonUtil.when(() -> CommonUtil.getUriByFilename(any(String.class)))
+                    .thenThrow(mockException);
+
+            // We invoke 'putExternalResources' twice to invalidate cache
+            // and have two identical exceptions which the equal content
+            final int numberOfRuns = 2;
+            final String[] configHashes = new String[numberOfRuns];
+            final String[] externalResourceHashes = new String[numberOfRuns];
+            for (int i = 0; i < numberOfRuns; i++) {
+                final Configuration config = new DefaultConfiguration("myConfig");
+                final PropertyCacheFile cache = new PropertyCacheFile(config, cacheFile.getPath());
+                cache.load();
+
+                configHashes[i] = cache.get(PropertyCacheFile.CONFIG_HASH_KEY);
+                assertWithMessage("Config hash key should not be null")
+                        .that(configHashes[i])
+                        .isNotNull();
+
+                final Set<String> nonExistentExternalResources = new HashSet<>();
+                final String externalResourceFileName = "non_existent_file.xml";
+                nonExistentExternalResources.add(externalResourceFileName);
+                cache.putExternalResources(nonExistentExternalResources);
+
+                externalResourceHashes[i] = cache.get(PropertyCacheFile.EXTERNAL_RESOURCE_KEY_PREFIX
+                        + externalResourceFileName);
+                assertWithMessage("External resource hashes should not be null")
+                        .that(externalResourceHashes[i])
+                        .isNotNull();
+
+                cache.persist();
+
+                final Properties cacheDetails = new Properties();
+                try (BufferedReader reader = Files.newBufferedReader(cacheFile.toPath())) {
+                    cacheDetails.load(reader);
+                }
+
+                final int expectedNumberOfObjectsInCacheFile = 2;
+                assertWithMessage("Unexpected number of objects in cache")
+                        .that(cacheDetails)
+                        .hasSize(expectedNumberOfObjectsInCacheFile);
+            }
+
+            assertWithMessage("Invalid config hash")
+                    .that(configHashes[0])
+                    .isEqualTo(configHashes[1]);
+            assertWithMessage("Invalid external resource hashes")
+                    .that(externalResourceHashes[0])
+                    .isEqualTo(externalResourceHashes[1]);
+        }
+    }
+
+    /**
+     * It is OK to have long test method name here as it describes the test purpose.
+     */
```

Comment:
May be use https://junit.org/junit5/docs/5.0.3/api/org/junit/jupiter/api/DisplayName.html instead? According to its doc, it is not its direct purpose, but it will be more readable

Reply:
This test was copied as is from `PropertyCacheFilePowermockTest.java`.
But you are right. The description should be in the Javadoc. I''l change the test name ant the comment.

Reply:
done. See below.

Reply:
In this case CsvSource annotation will be better than MethodSource

Reply:
> 

Please explain how to use `CsvSource` to produce array or collection of `String`?

Reply:
You cannot. 
I meant it may make sense to get rid of list and do it simple and stupid - make 2 args and use them with CsvSource.

Reply:
One boolean parameter? Yes, this will work. I'll update the PR.

Reply:
not sure what you mean, but will see :)